bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/transport/IpTransport.java

```diff
@@ -0,0 +1,414 @@
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.homekit.internal.session.AsymmetricSessionKeys;
+import org.openhab.binding.homekit.internal.session.EventListener;
+import org.openhab.binding.homekit.internal.session.HttpPayloadParser;
+import org.openhab.binding.homekit.internal.session.SecureSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This provides the IP transport layer for HomeKit communication.
+ * It provides methods for sending GET, POST, and PUT requests with appropriate headers and content types.
+ * It supports both plain and secure (encrypted) communication based on whether session keys have been set.
+ * It handles building HTTP requests, sending them over a socket, and parsing HTTP responses.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class IpTransport implements AutoCloseable {
+
+    private static final int TIMEOUT_MILLI_SECONDS = 15000; // HomeKit spec expects "around 10 seconds" so be safe
+    private static final Duration MINIMUM_REQUEST_INTERVAL = Duration.ofMillis(250);
+
+    private final Logger logger = LoggerFactory.getLogger(IpTransport.class);
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(r -> {
+        Thread t = new Thread(r, "homekit-io");
+        t.setDaemon(true);
+        return t;
+    });
+
+    private final Socket socket;
+    private final String hostName;
+    private final EventListener eventListener;
+
+    private volatile @Nullable SecureSession secureSession = null;
+    private volatile @Nullable Thread readThread = null;
+    private volatile @Nullable CompletableFuture<byte[][]> readHttpResponseFuture = null;
+
+    private boolean closing = false;
+    private Instant earliestNextRequestTime = Instant.MIN;
+
+    /**
+     * Creates a new IpTransport instance on the given host.
+     *
+     * @param ipAddress the IP address and port of the HomeKit accessory
+     * @param hostName the fully qualified host name (e.g. 'foobar._hap._tcp.local') of the HomeKit accessory
+     * @throws IOException
+     */
+    public IpTransport(String ipAddress, String hostName, EventListener eventListener) throws IOException {
+        logger.debug("Connecting to {} alias {}", ipAddress, hostName);
+        this.hostName = hostName;
+        this.eventListener = eventListener;
+        String[] parts = ipAddress.split(":");
+        socket = new Socket();
+        socket.setKeepAlive(true); // keep-alive forbidden for accessories but client should use it
+        socket.setTcpNoDelay(true); // disable Nagle algorithm to force immediate flushing of packets
+        socket.connect(new InetSocketAddress(parts[0], Integer.parseInt(parts[1])), TIMEOUT_MILLI_SECONDS);
```

Comment:
See other comment. Storing them separately as IP address (`text`) and port (`integer` with min/max) would eliminate the need to split the string and then parse the last part as integer.